File: src/main/java/org/intense/Ast/LinkingNode.java
package org.intense.Ast;

import org.intense.Environment;

import java.util.List;
import java.util.Map;

public class LinkingNode extends ASTNode{
    private final List<ASTNode> elements;
    /**
     * .b.c type is linking Node
     * primary functions are object life cycles
     * */
    public LinkingNode(List<ASTNode> elements) {
        this.elements = elements;
    }

    @Override
    public Object eval(Environment env) {
        return null;
    }
}

File: src/main/java/org/intense/ast/LinkingNode.kt
package org.intense.ast;

import org.intense.Env
import org.intense.Types.StrVal
import org.intense.Types.Value

class LinkingNode(instanceValue: String, element: MutableList<ASTNode>) : ASTNode(){
    private var instanceName = instanceValue;
    private var elements: List<ASTNode> = element
    /**
     * .b.c tokenType is linking Node
     * primary functions are object life cycles
     * */

    override fun eval(env: Env): Value {
        val output = StringBuilder()
        if (elements.isNotEmpty()) {
            for (it in elements)
            {
                //todo: add validation for class.objects
                output.append(it)
            }
        }
        return StrVal("$instanceName.$output")
    }
}
